### PR TITLE
chore: Do not expose getNodeBootstrappingPayload and getNodeBootstrappingCmd 

### PR DESCRIPTION
**Pull request by AbelHu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
1. Do not expose getNodeBootstrappingPayload and getNodeBootstrappingCmd since the rp will only call AgentBakerSVC via GetNodeBootstrapping.
2. Validate the input before calling other functions in GetNodeBootstrapping

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/datamodel/sig_config.go [hunk 1]
```diff
@@ -608,19 +608,22 @@ func GetSIGAzureCloudSpecConfig(sigConfig SIGConfig, region string) (SIGAzureEnv
 
 	fromACSUbuntu, err := withACSSIGConfig(sigConfig, "AKSUbuntu")
 	if err != nil {
-		return SIGAzureEnvironmentSpecConfig{}, fmt.Errorf("unexpected error while constructing env-aware sig configuration for AKSUbuntu: %s", err)
+		return SIGAzureEnvironmentSpecConfig{},
+			fmt.Errorf("unexpected error while constructing env-aware sig configuration for AKSUbuntu: %w", err)
```
Comment (AbelHu) [inline pkg/agent/datamodel/sig_config.go:612]: Updated `%s` to `%w` for fixing the golint error.

**Comment by alexeldeib** [comment]
overall lgtm, would you might breaking this up for reviewability? can ask some folks to help
- e2e switch to combined api
- fuzz switch to combined api
- fix lint issues
- remove/rename public funcs with no other change

cc @anujmaheshwari1 @cameronmeissner

**Comment by AbelHu** [comment]
> overall lgtm, would you might breaking this up for reviewability? can ask some folks to help
> 
> * e2e switch to combined api
> * fuzz switch to combined api
> * fix lint issues
> * remove/rename public funcs with no other change
> 
> cc @anujmaheshwari1 @cameronmeissner

@alexeldeib I have updated the PR per your comments.  Please help to review it again. The failed `Agentbaker E2E` should not be caused by this PR and it seems like to be a known issue.

cc @anujmaheshwari1 @cameronmeissner @junjiezhang1997 
image

File: e2e/scenarios/windows/property-windows-template.json [hunk 2]
```diff
@@ -69,8 +69,8 @@
         "windowsOffer": "aks-windows",
         "windowsPublisher": "microsoft-aks",
         "windowsSku": "",
-        "windowsPauseImageURL": "mcr.microsoft.com/oss/kubernetes/pause:3.6-hotfix.20220114",
-        "alwaysPullWindowsPauseImage": "false"
+        "windowsPauseImageURL": "mcr.microsoft.com/oss/kubernetes/pause:3.9",
```
Comment (alexeldeib) [inline e2e/scenarios/windows/property-windows-template.json:72]: is this related to the API changes? did tests break without this? what's the context?

File: fuzz/README.md [hunk 3]
```diff
@@ -10,11 +10,10 @@ This makes it a prime candidate for fuzz testing.
 
 ## Fuzz targets
 
-We currently have two fuzz targets, one for each major agentbaker API:
-- `customdata` fuzzes `baker.GetNodeBootstrappingPayload` to test custom data generation.
-- `csecmd` fuzzes `baker.GetNodeBootstrappingCmd` to test custom script command generation.
+We currently have one fuzz target for the major agentbaker API:
+- `api` fuzzes `agent.GetNodeBootstrapping` to test both custom data generation and custom script command generation.
 
-Each target generates random data as input and attempts to decode it as JSON into `NodeBootstrappingConfiguration`.
```
Comment (alexeldeib) [inline fuzz/README.md:17]: did this actually change?

File: fuzz/api/main.go [hunk 4]
```diff
@@ -1,6 +1,7 @@
-package csecmd
+package api
```
Comment (alexeldeib) [inline fuzz/api/main.go:1]: need to update https://github.com/Azure/AgentBaker/blob/master/.clusterfuzzlite/build.sh#L3

File: fuzz/README.md (hunk 3 above)
Comment (alexeldeib) [reply fuzz/README.md:17]: I get it, only one target now.

File: fuzz/api/main.go (hunk 4 above)
Comment (alexeldeib) [reply fuzz/api/main.go:1]: the doc there has info - 
https://github.com/Azure/AgentBaker/blob/master/.clusterfuzzlite/README.md
https://google.github.io/oss-fuzz/getting-started/new-project-guide/go-lang/#buildsh

the package name/func name just needs to match

**Comment by cameronmeissner** [comment]
> 

AB E2E should be working properly again as of around yesterday morning or so

**Comment by AbelHu** [comment]
Abandon this PR after [chore: fuzz GetNodeBootstrapping instead of internal AgentBaker functions by cameronmeissner · Pull Request #2936 · Azure/AgentBaker (github.com)](https://github.com/Azure/AgentBaker/pull/2936) and [chore: privatize internal agentbaker APIs by cameronmeissner · Pull Request #2935 · Azure/AgentBaker (github.com)](https://github.com/Azure/AgentBaker/pull/2935) are merged
